src/core/PathTracingSceneGenerator.js

```diff
@@ -176,9 +178,29 @@ export class PathTracingSceneGenerator {
 		// generate the geometry
 		const result = staticGeometryGenerator.generate( geometry );
 		const materials = result.materials;
+		let needsMaterialIndexUpdate = result.changeType !== NO_CHANGE || materialUuids === null || materialUuids.length !== length;
+		for ( let index = 0, length = materials.length; ( index < length ) && ! needsMaterialIndexUpdate; index ++ ) {
+
+			if ( materialUuids === null || materialUuids.length !== length ) {
+
+				needsMaterialIndexUpdate = true;
+				break;
+
+			}
+
+			const material = materials[ index ];
+			if ( material.uuid !== materialUuids[ index ] ) needsMaterialIndexUpdate = true;
+
+		}
+
+		materialUuids = materials.map( material => material.uuid );
```

Comment:
This can move inside the below condition since the uuids won't have changed otherwise.

```js
if ( needsMaterialIndexUpdate ) {

  updateMaterialIndexAttribute( geometry, materials, materials );
  materialUuids = materials.map( material => material.uuid );

}
```